### PR TITLE
fix: added permissions for lambda to attach lambda to the VPC.

### PR DESCRIPTION
**Pull request by GuptaNavdeep1983**
#2732 
Updated the code to add permissions for lambda required in case the lambda needs to be attached to the VPC. The issue was reported only for runner_syncer lambda but this issue will be seen for all lambda functions which needs to be attached to VPC.

### CODE REVIEW

File: modules/runner-binaries-syncer/runner-binaries-syncer.tf [hunk 1]
```diff
@@ -100,6 +100,13 @@ resource "aws_iam_role_policy" "lambda_logging" {
   })
 }
 
+resource "aws_iam_role_policy" "lambda_syncer_vpc" {
+  name = "${var.prefix}-lambda-syncer-vpc"
```
Comment (npalm) [inline modules/runner-binaries-syncer/runner-binaries-syncer.tf:104]: Can you add a conditional and only activate the policy if the lambda is attached to the vpc aka `lambda_subnet_ids` has more than zero elements.

**Review by npalm — CHANGES_REQUESTED** [review]
@GuptaNavdeep1983 thanks looks good, one small suggestion

File: modules/runner-binaries-syncer/runner-binaries-syncer.tf (hunk 1 above)
Comment (GuptaNavdeep1983) [reply modules/runner-binaries-syncer/runner-binaries-syncer.tf]: Addressed.

File: modules/runner-binaries-syncer/runner-binaries-syncer.tf [hunk 2]
```diff
@@ -100,6 +100,14 @@ resource "aws_iam_role_policy" "lambda_logging" {
   })
 }
 
+resource "aws_iam_role_policy" "lambda_syncer_vpc" {
+  count = var.lambda_subnet_ids != null && var.lambda_security_group_ids != null ? 1 : 0
```
Comment (npalm) [inline modules/runner-binaries-syncer/runner-binaries-syncer.tf:104]: This won't work would, default for both `[]`. So would check for a non empty list.

Comment (GuptaNavdeep1983) [reply modules/runner-binaries-syncer/runner-binaries-syncer.tf:104]: I have tested it and it worked just fine!

Comment (npalm) [reply modules/runner-binaries-syncer/runner-binaries-syncer.tf:104]: I testing the default example and it tries to create the policies, which should not be created



``` 
  # module.runners.module.runners.aws_iam_role_policy.lambda_scale_up_vpc[0] will be created
  + resource "aws_iam_role_policy" "lambda_scale_up_vpc" {
      + id     = (known after apply)
      + name   = "npalm-lambda-scale-up-vpc"
      + policy = jsonencode(
            {
              + Statement = [
                  + {
                      + Action   = [
                          + "ec2:CreateNetworkInterface",
                          + "ec2:DescribeNetworkInterfaces",
                          + "ec2:DeleteNetworkInterface",
                        ]
                      + Effect   = "Allow"
                      + Resource = "*"
                    },
                ]
              + Version   = "2012-10-17"
            }
        )
      + role   = "npalm-action-scale-up-lambda-role"
    }
```

**Review by npalm — APPROVED** [review]
Thanks @navdeepg2021 looks good

**Comment by npalm** [comment]
close #2732

**Pull request merged**